src/main/java/frc/robot/Robot.java

```diff
@@ -70,11 +75,23 @@ public void teleopInit() {
     if (m_autonomousCommand != null) {
       m_autonomousCommand.cancel();
     }
+    
   }
 
   /** This function is called periodically during operator control. */
+  PhotonVision camera = RobotContainer.camera;
   @Override
-  public void teleopPeriodic() {}
+  public void teleopPeriodic() {
+    
+    if(RobotContainer.xboxController.getAButton()){
+      camera.getCamera().setPipelineIndex(VisionConstants.kAprilPipline);
+      System.out.println("april tag pipeline");
+    }
+    else if(RobotContainer.xboxController.getBButton()){
+      camera.getCamera().setPipelineIndex(VisionConstants.kRetroPipline);
+      System.out.println("retroreflective pipline");
+    }
+  }
```

Comment:
Okay, so let's dive into WPILib for a second... What is the **command based programming** we use for our robot code? It is a code design in which describes every _physical mechanism_ in a digital object called a subsystem with certain capabilities like driving, climbing, shooting, etc... In order to activate those abilities we use a "Command" which is an object thats supposed to use the subsystem. These subsystem don't have to be physical as one can exist for PhotonVision (As you proved...).

What you did: Ignored the command part.
Button bindings should activate commands, and why? Activating commands enables WPILib to optimize the runtime of the code, optimize timing of commands and also makes the code more readable as all button related operations happen under the same roof of `RobotContainer`s `configureButtonBindings()` function.